projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,591 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## UI Design Guidance
+
+In this document, you will find sections titled **Design Bill** that outline requirements for UI screens and components. These sections provide guidelines while allowing designers room for creativity. A key principle to guide your work is **Clear Sign.** This principle emphasizes that information presented to users should be crafted with clear and knowledgeable intent. For more details, please refer to the article ["Clear-Sign Your Worries Away."](https://www.ledger.com/blog/clear-sign-your-worries-away)
+
+## Unleash
+[unleash]: #unleash
+
+First, we should enable the user's device in the `nano-contract.rollout` feature flag on Unleash. For this feature flag we use the default variant with 3 strategies:
+1. UserIds: this first strategy is dedicated for internal testers
+2. UserIds: this second strategy is dedicated for external testers
+3. Gradual rollout: this third strategy is dedicated to the public
+
+Each strategy [must](https://www.rfc-editor.org/rfc/rfc2119) support the following values:
+- `platform`: `ios`, `android`
+- `network`: `mainnet`, `testnet`, `privatenet`
+- `stage`: `mainnet`, `testnet`
+
+Second, we should enable the user's device in the `wallet-connect-mobile.rollout` feature flag on Unleash.
+
+## Opt-in
+
+```mermaid
+graph LR
+    A[Settings] --> B[Nano Contract Settings]
+```
+
+### Screen: Settings
+[screen-settings]: #screen-settings
+
+The "Settings" screen is divided into two sections: "General Settings" and "Dev Settings." Within "General Settings", the *Nano Contract* option is included.
+
+This option is visible only when the device is registered with Unleash. For registration and feature flag activation, see [Unleash](#unleash).
+
+Selecting the *Nano Contract* option initiates navigation to the *Nano Contract Settings* screen. This action allows users to modify their nano-contract preferences.
+
+##### Design Bill
+- *Nano Contract* option:
+	- **Placement**: Insert this option in the "General Settings" section, positioned before "Reset wallet" item.
+	
+	- **Functionality**: The option responds to touch input. A tap triggers a transition to the *Nano Contract Settings* screen.
+	
+	- **Visibility Conditions**: Display this option based on the device's registration status with Unleash.
+
+### Screen: Nano Contract Settings
+[screen-nano-contract-settings]: #screen-nano-contract-settings
+
+The *Nano Contract Settings* screen features a "Nano Contract" toggle button. Users can tap this button to activate the nano-contract feature. The toggle button should have a context description for what it actually does "Enables the view of a Nano Contract list, a list of transactions per Nano Contract and the possibility to register and unregister a Nano Contract".
```

Comment:
> In that case, the better UX for me would be to ask the user to configure his wallet before loading anything. With the proposed solution, we require the user to load his wallet fully, notice that nano transactions are not loaded, find the option in the settings, enable it, and then load everything again. This might lead to lot of questions and complaints that the nano transactions are not showing.

Let me break this down.

>In that case, the better UX for me would be to ask the user to configure his wallet before loading anything.

You mean during the wallet setup? After select import or create a wallet?

>[...] notice that nano transactions are not loaded

No, the nano contract transactions are being loaded all the time. I have tested it. What is not happening is the balance calculation per Nano Contract ID and other index activity that may need to happen related to Nano Contract.

>[...] find the option in the settings, enable it, and then load everything again.

Yes, this would be necessary. However, we can use the Push Notification strategy to draw user's attention by using a "pop-up" modal in the first initialization of the wallet after it be allowed to access the feature. We can even improve this process by driven the user to an onboarding experience.

>[...] This might lead to lot of questions and complaints that the nano transactions are not showing.

Nano Contract transactions are already being shown in the wallet _as is_. By enabling the feature the user would only receive a better visualization, a more contextualized data and the option to engage with Nano Contract by using the Wallet Connect. This last topic is more problematic, indeed. However, I believe in the opposite effect, if we rollout the feature without user participation, it will expose the user to new capabilities as an unexpected move, and the consequence would be the doubt and complaints. On the other hand, if we give users the power to decide, we also gain a change to guide the process using an onboarding experience, and this can greatly remove noise.

>I prefer that we have the feature toggle in the unleash only and discuss this new feature of having a full configuration of the wallet in the future.

Give me your vision about the previous reply first. But sure, we can turn this feature toggle a "Future Possibility".